src/types/native.rs

```diff
@@ -14,6 +15,7 @@ pub trait NativeType:
     + Send
     + Sync
     + Sized
+    + RefUnwindSafe
```

Comment:
This is necessary to ensure that `arrow2::Bytes<T>` is `RefUnwindSafe` which is important to `arrow::Buffer`